app.js

```diff
@@ -109,85 +109,79 @@ var combineQueries = (q1, q2) => {
   return combinedQueryString;
 };
 
-var lookup = (slug, idOnly) => {
-  return new Promise(function (resolve, reject) {
-    const timeNow = Math.round(new Date().getTime() / 1000);
-
-    if (cache[slug] && timeNow < cache[slug].expires) {
-      // valid cache
-      console.log("Yeet. Cache has what I needed.");
-      console.log(cache[slug]);
-      resolve(idOnly ? cache[slug].id : cache[slug].dest);
+const lookup = async (slug, idOnly) => {
+  const timeNow = Math.round(new Date().getTime() / 1000);
+
+  if (cache[slug] && timeNow < cache[slug].expires) {
+    console.log("Yeet. Cache has what I needed.");
+    console.log(cache[slug]);
+    return idOnly ? cache[slug].id : cache[slug].dest;
+  } else {
+    console.log("Oops. Can't find useful data in cache. Asking Postgres.");
+    const res = await client.query(
+      "SELECT id, destination FROM links WHERE slug = $1",
```

Comment:
aren't queries like this vulnerable to SQL injection?

Reply:
No. Parameterized queries are safe on node-pg.